internal/agent/sender/rest.go

```diff
@@ -0,0 +1,51 @@
+package sender
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+)
+
+type RestSender struct {
+	addr string
+
+	// we use custom http client here for further customization
+	// and to enable connection reuse for sequential server calls
+	cl *http.Client
+}
+
+func NewRestSender(addr string) *RestSender {
+	return &RestSender{
+		addr: addr,
+		cl:   &http.Client{},
+	}
+}
+
+func (rs RestSender) MetricSendFunc() MetricSendFunc {
+	return func(id string, mtype string, value string) error {
+		req, err := http.NewRequest(http.MethodPost, rs.composePostMetricURL(id, mtype, value), nil)
+		if err != nil {
+			panic(err)
+		}
+
+		req.Header.Add("Content-Type", "text/plain")
+		resp, err := rs.cl.Do(req)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return err
```

Comment:
Ошибки из сторонних пакетов стоит оборачивать, чтобы было понятнее, что именно пошло не так:

```suggestion
			return fmt.Errorf("failed to read response body: %w", err)
```

Reply:
заимпрувил везде где можно было